Algorithms/Balanced_Parenthesis.cpp

```diff
@@ -0,0 +1,68 @@
+#include <iostream>
+#include <stack>
+#include <string>
+using namespace std;
+
+// Function to check if character x is an opening parenthesis and character y is an closing parenthesis or character x is 'less than' and character y is 'greater than' operator
+bool checkCharacter(char x, char y) 
+{
+	if ((x=='(' && y==')') || (x=='{' && y=='}') || (x=='[' && y==']') || (x=='<' && y=='>')) // compare characters
+	{
+		return true;
+	}
+	else
+	{
+		return false;
+	}
+}
+
+bool checkBalancedParenthesis(string s) // Function to check if the string has balanced paranthesis or not
+{
+	stack<char> st; // character stack to store given string characters
+	int top=-1;     // initialize top as -1 as the stack is 0 indexed
+	int l=s.length();
+	for(int i=0;i<l;i++)
+	{
+		if(s[i]=='(' || s[i]=='{' || s[i]=='[') // if the character of the string is opening paranthesis, push it to stack
+		{
+		st.push(s[i]);
+		top=top+1;   // increament top by 1 as character is pushed
+		}
+		else if(s[i]==')' || s[i]=='}' || s[i]==']')  // if character is closing parenthesis, compare with stack top
+		{
+			if(top==-1 || checkCharacter(st.top(),s[i])==false) // if no element in stack or character at top does not match with corresponding current character
+			{
+				return false;
+			}
+			else
+			{
+			st.pop(); // if matches , pop the top of stack 
```

Comment:
Please read the [Coding Standards](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards) wiki page and write code with proper indentation. Follow all the guidelines given there.